tasks/cops_documentation.rake

```diff
@@ -10,291 +11,21 @@ end
-    File.open(file_name, 'w') do |file|
-      puts "* generated #{file_name}"
-      file.write(content.strip + "\n")
-    end
-  end
-
-  def print_cop_with_doc(cop, config)
-    t = config.for_cop(cop)
-    non_display_keys = %w[
-      Description Enabled StyleGuide Reference Safe SafeAutoCorrect VersionAdded
-      VersionChanged
-    ]
-    pars = t.reject { |k| non_display_keys.include? k }
-    description = 'No documentation'
-    examples_object = []
-    cop_code(cop) do |code_object|
-      description = code_object.docstring unless code_object.docstring.blank?
-      examples_object = code_object.tags('example')
-    end
-    cops_body(config, cop, description, examples_object, pars)
-  end
-
-  def cop_code(cop)
-    YARD::Registry.all(:class).detect do |code_object|
-      next unless RuboCop::Cop::Badge.for(code_object.to_s) == cop.badge
-
-      yield code_object
-    end
-  end
-
-  def table_of_content_for_department(cops, department)
-    type_title = department[0].upcase + department[1..-1]
-    filename = "cops_#{department.downcase}.adoc"
-    content = +"=== Department xref:#{filename}[#{type_title}]\n\n"
-    cops_of_department(cops, department.to_sym).each do |cop|
-      anchor = cop.cop_name.sub('/', '').downcase
-      content << "* xref:#{filename}##{anchor}[#{cop.cop_name}]\n"
-    end
-
-    content
-  end
-
-  def print_table_of_contents(cops)
-    path = "#{Dir.pwd}/docs/modules/ROOT/pages/cops.adoc"
-    original = File.read(path)
-    content = +"// START_COP_LIST\n\n"
-
-    content << table_contents(cops)
-
-    content << "\n// END_COP_LIST"
-
-    content = original.sub(
-      %r{// START_COP_LIST.+// END_COP_LIST}m, content
-    )
-    File.write(path, content)
-  end
-
-  def table_contents(cops)
-    cops
-      .departments
-      .map(&:to_s)
-      .sort
-      .map { |department| table_of_content_for_department(cops, department) }
-      .join("\n")
-  end
-
-  def cop_status(status)
-    return 'Disabled' unless status
-
-    status == 'pending' ? 'Pending' : 'Enabled'
-  end
-
-  def assert_docs_synchronized
-    # Do not print diff and yield whether exit code was zero
-    sh('git diff --quiet docs') do |outcome, _|
-      return if outcome
-
-      # Output diff before raising error
-      sh('GIT_PAGER=cat git diff docs')
-
-      warn 'The docs directory is out of sync. ' \
-        'Run `rake generate_cops_documentation` and commit the results.'
-      exit!
-    end
-  end
-
-  def main
-    cops   = RuboCop::Cop::Cop.registry
-    config = RuboCop::ConfigLoader.default_configuration
+  deps = %w[Bundler Gemspec Layout Lint Metrics Migration Naming Security Style]
+  CopsDocumentationGenerator.new(departments: deps).call
+end
 
-    YARD::Registry.load!
-    cops.departments.sort!.each do |department|
-      print_cops_of_department(cops, department, config)
-    end
+desc 'Confirm that documentation is up to date'
+task confirm_cops_documentation: :generate_cops_documentation do
```

Comment:
I'd probably use "verify" or "check" here.

Reply:
Changed to `verify_cops_documentation`